src/main/java/edu/arizona/kra/negotiations/dao/ojb/NegotiationLookupDaoOjb.java

```diff
@@ -0,0 +1,284 @@
+    }
+
+    /**
+     * Parses the user input for negotiation Age and returns the appropriate Criteria for it.
+     * If the input is empty or cannot be parsed, will return an empty Criteria.
+     *
+     * @param ageFilter Valid examples: integer, >minvalue, <maxvalue, minvalue..maxvalue
+     * @return
+     */
+    private Criteria getNegotiationAgeCriteria( String ageFilter ) {
+        if ( StringUtils.isEmpty(ageFilter) ) {
+            LOG.debug("getCriteria: No additional filters for ageFilter.");
+            return new Criteria();
+        }
+        Criteria negotiationAgeCriteria = new Criteria();
+        LOG.debug("Add negotiation age filter={}", ageFilter);
+
+        //Parse the ageFilter to find out what operation we need and what are the limits
+        //TODO refactor this old code...
+        int lowValue = 0;
+        int highValue = 0;
+        int equalValue = 0;
+        boolean greaterThan = false;
+        boolean lessThan = false;
+        boolean between = false;
+
+        try {
+            if (ageFilter.contains(">")) {
+                greaterThan = true;
+                lowValue = Integer.parseInt(ageFilter.replace(">", ""));
+            } else if (ageFilter.contains("<")) {
+                lessThan = true;
+                highValue = Integer.parseInt(ageFilter.replace("<", ""));
+            } else if (ageFilter.contains("..")) {
+                between = true;
+                String[] values = ageFilter.split("\\.\\.");
+                lowValue = Integer.parseInt(values[0]);
+                highValue = Integer.parseInt(values[1]);
+            } else {
+                equalValue = Integer.parseInt(ageFilter);
+            }
+        } catch (NumberFormatException e) {
+            LOG.warn("Negotiation age criteria unParsable={} ignoring...", ageFilter, e);
+            return negotiationAgeCriteria;
+        }
+
+        //create subquery for the associated object with the above criteria
+        StringBuffer negotiationAgeSqlQuery = new StringBuffer( NEGOTIATION_AGE_QUERY );
+        if ( equalValue != 0){
+            negotiationAgeSqlQuery.append( NEGOTIATION_AGE_CONDITION + "=" + equalValue );
+        } else if ( greaterThan ){
+            negotiationAgeSqlQuery.append( NEGOTIATION_AGE_CONDITION + ">=" + lowValue );
+        } else if ( lessThan ){
+            negotiationAgeSqlQuery.append( NEGOTIATION_AGE_CONDITION + "<=" + highValue );
+        } else if ( between ){
+            negotiationAgeSqlQuery.append( NEGOTIATION_AGE_CONDITION + ">=" + lowValue );
+            negotiationAgeSqlQuery.append( " AND ");
+            negotiationAgeSqlQuery.append( NEGOTIATION_AGE_CONDITION + "<=" + highValue);
+        } else {
+            LOG.warn("Negotiation age criteria: Uncharted territory!!! {} Ignoring...", ageFilter);
+            return negotiationAgeCriteria;
+        }
+
+        QueryBySQL sqlSubQuery = QueryFactory.newQuery(Negotiation.class, negotiationAgeSqlQuery.toString());
```

Comment:
That is correct, we're utilizing OJB as ORM, however, I couldn't find a way to express the query for negotiation age in ORM terms. In the DB, we only keep start date and end date so the query would have to calculate on spot the difference between the two, keeping in mind that the end date could be empty, in which case it has to consider the current timestamp.